[commits.webkit.org] Individually push commits to mirrored repositories

JonWBedard opened the pull request:
#### c5a6f77e96c84a92eb3983a3203dafd70bc4385f
<pre>
[commits.webkit.org] Individually push commits to mirrored repositories
<a href="https://bugs.webkit.org/show_bug.cgi?id=266595">https://bugs.webkit.org/show_bug.cgi?id=266595</a>
<a href="https://rdar.apple.com/119830188">rdar://119830188</a>

Reviewed by Alexey Proskuryakov.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py:
(Checkout.push_update): If less than 50 commits are being synced, push them individually.

Canonical link: <a href="https://commits.webkit.org/272244@main">https://commits.webkit.org/272244@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/081240889f1dee96438dcccd171242e55e7d5877

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31064 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32753 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33575 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/28052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31827 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12083 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6998 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27890 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31396 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/8209 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/27782 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/7043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/31238 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/7219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/27664 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34913 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/28280 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/28132 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/33349 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/7254 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/5318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/31182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/8946 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/7957 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4032 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/7793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py:
```diff
@@ -237,6 +237,23 @@ def push_update(self, branch=None, tag=None, remote=None, track=False, dest_bran
 
+        # Make an attempt to push commits one-at-a-time to ensure we get CI jobs for each commit
+        # Only push commits one-at-a-time if we have less than 50 individual commits.
+        sentinal = run(
```
typo: should be "sentinel"

aproskuryakov commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py:
```diff
@@ -237,6 +237,23 @@ def push_update(self, branch=None, tag=None, remote=None, track=False, dest_bran
             return False
 
+        # Make an attempt to push commits one-at-a-time to ensure we get CI jobs for each commit
+        # Only push commits one-at-a-time if we have less than 50 individual commits.
```
Seems worth mentioning the `main` special case here and in the commit message.

aproskuryakov reviewed (approved):
I wish we could have unit tests for such behaviors.

JonWBedard committed:
[commits.webkit.org] Individually push commits to mirrored repositories
https://bugs.webkit.org/show_bug.cgi?id=266595
rdar://119830188

Reviewed by Alexey Proskuryakov.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py:
(Checkout.push_update): If less than 50 commits are being synced, push them individually.

Canonical link: https://commits.webkit.org/272244@main
